generate.ml

```diff
@@ -146,11 +146,38 @@ let gen_const typ =
   | _ ->
     Cil.mkCast ~force:false ~e:(Cil.zero ~loc) ~newt:typ
 
+let gen_array_elt (lhost, offset as lval) =
+  match Cil.typeOfLval lval with
+  | TArray (_, e, _, _) ->
+    (* Generate random array index in [0, array_length - 1], just for the last
+       array dimension. *)
+    let offset =
+      match offset with
+      | NoOffset ->
+        Index (Cil.(integer ~loc (Random.int (lenOfArray e))), NoOffset)
+      | _ ->
+        offset
```

Comment:
Can this case with an existing offset ever happen?

Reply:
You already know the answer :)

The answer is no, at least at the moment. But this is not really a problem in my opinion, or do you prefer an `assert false` to be sure that future developments don't forget the case?

Reply:
Yes, please add an `assert`.